.github/workflows/test_and_deploy.yml

```
@@ -75,7 +75,7 @@ jobs:
           TOX_SKIP_ENV: ".*py39-PySide6.*"
 
       - name: Coverage
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
```

We might want to stick to v3 if there's no token around. 

Or instead:

```suggestion
        uses: codecov/codecov-action@v4
        env:
          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

---

v3 has big problems now. We should migrate to v4 and add token

---

Is the token an org token or a repo token? Can we get one added here if needed? Thanks 🙏